components/reusable-components/text-input/TextInput.tsx

```diff
@@ -0,0 +1,63 @@
+'use client';
+
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import React, { FC, HTMLProps } from 'react';
+import style from './textInput.module.scss';
+
+interface InputProps extends HTMLProps<HTMLInputElement> {
+  placeholder: string;
+  label: string;
+  name: string;
+}
+
+const TextInput: FC<InputProps> = ({ placeholder, label, name }) => {
+  const inputRegex = /^[\u0400-\u04FFа-џA-Za-z-]+$/;
+
+  const formik = useFormik({
+    initialValues: { [name]: '' },
+    validationSchema: Yup.object({
+      name: Yup.string()
+        .required('Полето е задолжително')
+        .matches(
+          inputRegex,
+          '*Дозволени се следните карактери.... името може да биде на кирилица и латиница'
+        ),
```

Comment:
This looks good so far.
This input will need to accept:

- validation schema
- placeholder
- error message?
- the input type can be: text, email, number
- name would be also dynamically added
-

Reply:
Also the input should be able to accept props, We might need to use for example: disabled